src/vs/editor/contrib/hover/browser/resizableContentWidget.ts

```diff
@@ -0,0 +1,122 @@
+		this._resizableNode.layout(_initialSize.height, _initialSize.width);
+		this._register(this._resizableNode.onDidResize(e => {
+			this._resize(new dom.Dimension(e.dimension.width, e.dimension.height));
+			if (e.done) {
+				this._isResizing = false;
+			}
+		}));
+		this._register(this._resizableNode.onDidWillResize(() => {
+			this._isResizing = true;
+		}));
+	}
+
+	get isResizing() {
+		return this._isResizing;
+	}
+
+	abstract getId(): string;
+
+	getDomNode(): HTMLElement {
+		return this._resizableNode.domNode;
+	}
+
+	getPosition(): IContentWidgetPosition | null {
+		return this._contentPosition;
+	}
+
+	get position(): Position | undefined {
+		return this._position;
+	}
+
+	set position(position: Position | undefined) {
+		this._position = position;
+	}
+
+	protected _availableVerticalSpaceAbove(position: IPosition): number | undefined {
+		const editorDomNode = this._editor.getDomNode();
+		const mouseBox = this._editor.getScrolledVisiblePosition(position);
+		if (!editorDomNode || !mouseBox) {
+			return;
+		}
+		const editorBox = dom.getDomNodePagePosition(editorDomNode);
+		return editorBox.top + mouseBox.top - HEADER_HEIGHT;
+	}
+
+	protected _availableVerticalSpaceBelow(position: IPosition): number | undefined {
+		const editorDomNode = this._editor.getDomNode();
+		const mouseBox = this._editor.getScrolledVisiblePosition(position);
+		if (!editorDomNode || !mouseBox) {
+			return;
+		}
+		const editorBox = dom.getDomNodePagePosition(editorDomNode);
+		const bodyBox = dom.getClientArea(document.body);
+		const mouseBottom = editorBox.top + mouseBox.top + mouseBox.height;
+		return bodyBox.height - mouseBottom;
+	}
+
+	protected _findPositionPreference(widgetHeight: number, showAtPosition: IPosition): ContentWidgetPositionPreference | undefined {
+		const maxHeightBelow = Math.min(this._availableVerticalSpaceBelow(showAtPosition) ?? Infinity, widgetHeight);
+		const maxHeightAbove = Math.min(this._availableVerticalSpaceAbove(showAtPosition) ?? Infinity, widgetHeight);
+		const maxHeight = Math.min(Math.max(maxHeightAbove, maxHeightBelow), widgetHeight);
+		const height = clamp(widgetHeight, MIN_HEIGHT, maxHeight);
+		let renderingAbove: ContentWidgetPositionPreference;
+		if (this._editor.getOption(EditorOption.hover).above) {
+			renderingAbove = height <= maxHeightAbove ? ContentWidgetPositionPreference.ABOVE : ContentWidgetPositionPreference.BELOW;
+		} else {
+			renderingAbove = height <= maxHeightBelow ? ContentWidgetPositionPreference.BELOW : ContentWidgetPositionPreference.ABOVE;
+		}
+		if (renderingAbove === ContentWidgetPositionPreference.ABOVE) {
+			this._resizableNode.enableSashes(true, true, false, false);
+		} else {
+			this._resizableNode.enableSashes(false, true, true, false);
+		}
+		return renderingAbove;
+	}
+
+	_resize(dimension: dom.Dimension): void {
+		this._resizableNode.layout(dimension.height, dimension.width);
+	}
+
+	beforeOnDidWillResize() {
+		return;
+	}
+
+	afterOnDidResize() {
+		return;
+	}
```

Comment:
these never seem to be used (no caller, no implementor)

Reply:
That's right, this is code from the initial PR which persists the hover sizes.